Tutorials/OpenData/flowAnalysis.cxx

```diff
@@ -0,0 +1,375 @@
+    LOGF(info, "Number of tracks: %d", tracks.size());
+    */
+
+    // process the tracks of a given collision
+    Double_t QxnGapA = 0., QynGapA = 0.;
+    Double_t QxnGapC = 0., QynGapC = 0.;
+
+    Int_t multGapA = 0, multGapC = 0;
+
+    // Tracks are already filtered with GlobalTrack || GlobalTrackSDD
+    Int_t multTrk = tracks.size();
+
+    for (auto& track : tracks) {
+
+      if (TMath::Abs(track.eta()) >= etaCut ||
+          track.tpcNClsFound() < noClus ||
+          track.pt() < minPt || track.pt() >= maxPt)
+        continue;
+
+      if (track.eta() > etaGap) {
+        QxnGapC += TMath::Cos(nHarm * track.phi());
+        QynGapC += TMath::Sin(nHarm * track.phi());
+        multGapC++;
+      }
+
+      if (track.eta() < -etaGap) {
+        QxnGapA += TMath::Cos(nHarm * track.phi());
+        QynGapA += TMath::Sin(nHarm * track.phi());
+        multGapA++;
+      }
+    }
+
+    histos.fill(HIST("multvsCent"), v0Centr, multTrk);
+
+    if (multGapA > 0 && multGapC > 0) {
+      Double_t resGap = (QxnGapA * QxnGapC + QynGapA * QynGapC) / (Double_t)multGapA / (Double_t)multGapC;
+      histos.fill(HIST("res"), v0Centr, resGap);
+
+      histos.fill(HIST("QxnA"), v0Centr, QxnGapA / (Double_t)multGapA);
+      histos.fill(HIST("QxnC"), v0Centr, QxnGapC / (Double_t)multGapC);
+
+      histos.fill(HIST("QynA"), v0Centr, QynGapA / (Double_t)multGapA);
+      histos.fill(HIST("QynC"), v0Centr, QynGapC / (Double_t)multGapC);
+    }
+
+    for (auto& track : tracks) {
+
+      if (TMath::Abs(track.eta()) >= etaCut ||
+          track.tpcNClsFound() < noClus ||
+          track.pt() < minPt || track.pt() >= maxPt)
+        continue;
+
+      if (track.eta() > etaGap && multGapA > 0) {
+        Double_t vnC = (TMath::Cos(nHarm * track.phi()) * QxnGapA +
+                        TMath::Sin(nHarm * track.phi()) * QynGapA) /
+                       (Double_t)multGapA;
+        // HistogramRegistry does not support access via non-const variables...
+        switch (centrCode) {
+          case 0: {
+            fillCPt<0>(track, vnC);
+            break;
+          }
+          case 1: {
+            fillCPt<1>(track, vnC);
+            break;
+          }
+          case 2: {
+            fillCPt<2>(track, vnC);
+            break;
+          }
+          case 3: {
+            fillCPt<3>(track, vnC);
+            break;
+          }
+          case 4: {
+            fillCPt<4>(track, vnC);
+            break;
+          }
+          case 5: {
+            fillCPt<5>(track, vnC);
+            break;
+          }
+          case 6: {
+            fillCPt<6>(track, vnC);
+            break;
+          }
+          case 7: {
+            fillCPt<7>(track, vnC);
+            break;
+          }
+          case 8: {
+            fillCPt<8>(track, vnC);
+            break;
+          }
+          case 9: {
+            fillCPt<9>(track, vnC);
+            break;
+          }
+        }
```

Comment:
It is up to you, but here maybe you could consider having the centrality as one additional dimension to the vn histograms, so you could avoid this whole switch statement.

Reply:
Thanks for the hint @iarsene, it is done like this now.